app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,67 @@ def show
     @post = Post.find(params[:id])
     @user = User.find(params[:user_id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+    @id = params[:user_id]
+  end
+
+  def create
+    @post = Post.create(post_params)
+    @user = current_user
+    @post.author_id = @user.id
+
+    # respond_to block
+    respond_to do |f|
+      f.html do
+        if @post.save
+          flash[:success] = 'Post created successfully!'
+          redirect_to "/users/#{@user.id}/posts/"
+        else
+          flash.now[:error] = 'Error creating post!'
+          render :new
+        end
+      end
+    end
+  end
+
+  # Create comment
+  def comment_post
+    # new object from params
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = Post.find(params[:id])
+
+    if @comment.save
+      flash[:success] = 'Comment created successfully!'
+      redirect_to user_post_path
+    else
+      flash.now[:error] = 'Error creating comment!'
+      render :new
+    end
+  end
+
+  def like_post
+    @post = Post.find(params[:id])
+    # new object from params
+    @like = Like.new(author_id: current_user.id, post_id: @post.id)
+    if @like.save
+      flash[:success] = 'You liked the post!'
+      redirect_to user_post_path
+    else
+      flash.now[:error] = 'Error: Post could not be liked'
+    end
+  end
```

Comment:
- [ ] Please keep in mind to put this method in the `likes_controller`

Finally, please verify that a user can still perform the `post creation`, `commenting` and `liking` actions successfully before asking for a re-review.

Reply:
Hello @mmsesay Thank you for clarifying me the suggested changes!
i am already working on them, they will soon be implemented!
Have a good night @DeliceLydia , @mmsesay

Reply:
You are welcome @Pamphilemkp. Good night! 🌃